packages/block-library/src/template-part/edit/index.js

```diff
@@ -80,10 +86,16 @@ function TemplatesList( { area, clientId, isEntityAvailable, onSelect } ) {
 	// This hook fetches patterns, so don't run it unconditionally in the main
 	// edit function!
 	const blockPatterns = useAlternativeBlockPatterns( area, clientId );
+	const isOverlayExperimentEnabled =
+		typeof window !== 'undefined' &&
+		window.__experimentalNavigationOverlays === true;
+	const supportedAreas = isOverlayExperimentEnabled
+		? [ ...SUPPORTED_AREAS, 'overlay' ]
+		: SUPPORTED_AREAS;
```

Comment:
Can we abstract this out to avoid the repetition?

Reply:
this will go away when we remove the experiment, does it make sense to add more complexity?

Reply:
No strong feelings either way, but if we do it in a single place we'll have less code to remove when removing the experiment.